src/server/search_family.cc

```diff
@@ -0,0 +1,252 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "server/search_family.h"
+
+#include <variant>
+#include <vector>
+
+#include "base/logging.h"
+#include "core/search/interface.h"
+#include "facade/reply_builder.h"
+#include "server/command_registry.h"
+#include "server/conn_context.h"
+#include "server/container_utils.h"
+#include "server/engine_shard_set.h"
+#include "server/transaction.h"
+
+extern "C" {
+#include "redis/listpack.h"
+#include "redis/object.h"
+};
+
+namespace dfly {
+
+using namespace std;
+using namespace facade;
+
+namespace {
+
+using SerializedDocument = absl::flat_hash_map<std::string, std::string>;
+
+string_view SdsToSafeSv(sds str) {
+  if (str == nullptr)
+    return ""sv;
+
+  return string_view{str, sdslen(str)};
+}
+
+struct BaseAccessor : public search::HSetAccessor {
+  using FieldConsumer = search::HSetAccessor::FieldConsumer;
+
+  virtual SerializedDocument Serialize() const = 0;
+};
+
+struct ListPackAccessor : public BaseAccessor {
+  using LpPtr = uint8_t*;
+
+  ListPackAccessor(LpPtr ptr = nullptr) : lp_{ptr} {
+  }
+
+  bool Check(FieldConsumer f, string_view active_field) const override {
```

Comment:
Should this be `const FieldConsumer&`?

Reply:
Its taken by value inside the base class, so I kept it